packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -57,6 +51,43 @@ export function isQueryFieldsValid(fields: Record<string, QueryBarFormField>) {
   );
 }
 
+function wrapFilterIfNotWrapped(
+  filter: QueryBarState['fields']['filter']
+): QueryBarState['fields']['filter'] {
+  if (validateField('filter', filter.string) !== false || !filter) {
```

Comment:
Nit: you usually want to do a fast check first so that it can stop the expression fast and avoid moving to the next part of the expression. In this case checking whether or not filter was passed first means you don't need to run validate code it if it isn't